manifests/init.pp

```diff
@@ -86,12 +87,21 @@
   $pool_test_on_borrow                                              = false,
   # JVM Settings
   $javahome                                                         = undef,
-  $jvm_xms                                                          = '256m',
-  $jvm_xmx                                                          = '1024m',
-  $jvm_permgen                                                      = '256m',
-  $jvm_optional                                                     = '-XX:-HeapDumpOnOutOfMemoryError',
-  $java_opts                                                        = '',
-  $catalina_opts                                                    = '',
+  Jira::Jvm_types $jvm_type                                         = 'oracle-jdk-1.8',
+  String $jvm_xms                                                   = '256m',
+  String $jvm_xmx                                                   = '1024m',
+  String $jvm_permgen                                               = '256m',
+  Optional[String] $jvm_optional                                    = '',
```

Comment:
If the datatype is `Optional[]`, the value also needs to be `undef`. You cannot pass `undef` as a value to the jira class. Please switch the datatype to `String` or stick with `Optional` but change the default value from empty string (`''`) to `undef`.

Reply:
Good catch, fixed.